test/functional/tests/fault_injection/test_fault_injection_with_mounted_core.py

```diff
@@ -166,3 +166,53 @@ def test_add_occupied_core(cache_mode):
 
     with TestRun.step("Stop all caches."):
         casadm.stop_all_caches()
+
+
+@pytest.mark.parametrize("cache_mode", CacheMode)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_add_twice_same_core(cache_mode):
+    """
+        title: Fault injection test if cache does not accept twice the same device as core.
+        description: |
+          Negative test of the ability to add the same core twice to cache.
+        pass_criteria:
+          - No system crash.
+          - The same core device cannot be used twice in cache.
+    """
+    with TestRun.step("Prepare cache and core devices."):
+        cache_dev = TestRun.disks['cache']
+        cache_dev.create_partitions([Size(512, Unit.MebiByte), Size(512, Unit.MebiByte)])
+        cache_dev1 = cache_dev.partitions[0]
+        cache_dev2 = cache_dev.partitions[1]
+        core_dev = TestRun.disks['core']
+        core_dev.create_partitions([Size(2, Unit.GibiByte)])
+        core_dev = core_dev.partitions[0]
+
+    with TestRun.step("Start two caches"):
+        cache1 = casadm.start_cache(cache_dev1, cache_mode, force=True)
+        cache2 = casadm.start_cache(cache_dev2, cache_mode, force=True)
+
+    with TestRun.step("Add core device to cache."):
+        cache1.add_core(core_dev)
+
+    with TestRun.step("Try add second time the same core device to the same cache"):
+        output = TestRun.executor.run_expect_fail(cli.add_core_cmd(cache_id=str(cache1.cache_id),
+                                                                   core_dev=str(core_dev)))
+        if output.stderr:
```

Comment:
consider checking error message. There is a method to do so and it's called check_msg(). It is located in cli_messages.py